libs/ledger-live-common/src/families/vechain/utils/transaction-utils.ts

```diff
@@ -0,0 +1,155 @@
+  if (!randBuffer) throw Error("Failed to generate random hex");
+  return `${HEX_PREFIX}${randBuffer.toString("hex").substring(0, 8)}`;
+};
+
+/**
+ * Estimate the gas that will be used by the transaction.
+ * @param transaction - The transaction to estimate the gas for
+ * @returns an estimate of the gas usage
+ */
+export const estimateGas = async (t: Transaction): Promise<number> => {
+  const intrinsicGas = ThorTransaction.intrinsicGas(t.body.clauses);
+
+  const queryData: Query[] = [];
+
+  t.body.clauses.forEach(c => {
+    if (c.to) {
+      queryData.push({
+        to: c.to,
+        data: c.data,
+      });
+    }
+  });
+
+  const response = await query(queryData);
+
+  const execGas = response.reduce((sum, out) => sum + out.gasUsed, 0);
+
+  return intrinsicGas + (execGas ? execGas + GAS_COEFFICIENT : 0);
+};
+
+const getBaseGasPrice = async (): Promise<string> => {
+  const queryData: Query = {
+    to: PARAMS_ADDRESS,
+    data: params.get.encode(BASE_GAS_PRICE_KEY),
+  };
+
+  const response = await query([queryData]);
+
+  // Expect 1 value
+  if (response && response.length != 1) throw Error("Unexpected response received for query");
+
+  return response[0].data;
+};
+
+/**
+ * Calculate the fee in VTHO
+ * @param gas - the gas used
+ * @param gasPriceCoef - the gas price coefficient
+ * @returns the fee in VTHO
+ */
+export const calculateFee = async (gas: BigNumber, gasPriceCoef: number): Promise<BigNumber> => {
+  const baseGasPrice = await getBaseGasPrice();
+  return new BigNumber(baseGasPrice).times(gasPriceCoef).idiv(255).plus(baseGasPrice).times(gas);
+};
+
+export const calculateTransactionInfo = async (
+  account: Account,
+  transaction: Transaction,
+  fixedMaxTokenFees?: BigNumber,
+): Promise<TransactionInfo> => {
+  const { subAccounts } = account;
+  const { amount: oldAmount, useAllAmount, subAccountId } = transaction;
+  const maxTokenFees = fixedMaxTokenFees || (await calculateMaxFeesToken());
+
+  const tokenAccount =
+    subAccountId && subAccounts
+      ? (subAccounts.find(subAccount => {
+          return subAccount.id === subAccountId;
+        }) as TokenAccount)
+      : undefined;
+  const isTokenAccount = !!tokenAccount;
+
+  let balance;
+  let spendableBalance;
+  let amount;
+
+  if (isTokenAccount) {
+    balance = tokenAccount.balance;
+    spendableBalance = tokenAccount.balance.minus(maxTokenFees).gt(0)
+      ? tokenAccount.balance.minus(maxTokenFees)
+      : new BigNumber(0);
+    amount = useAllAmount ? spendableBalance : oldAmount;
+  } else {
+    balance = account.balance;
+    spendableBalance = account.balance;
+    amount = useAllAmount ? spendableBalance : oldAmount;
+  }
+
+  return {
+    isTokenAccount,
+    amount,
+    spendableBalance,
+    balance,
+    tokenAccount,
+  };
+};
+
+export const calculateMaxFeesToken = async (): Promise<BigNumber> => {
+  //FIXME: The BE call is currently returning not stable values, hardwiring a value
```

Comment:
Will it ever be fixed ? I think we need to choose between dynamic value and static value as it is implemented here

Reply:
asking it, thanks